File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.java
class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {
        int res_a=1,res_d=2;
        int l,w;
        for(int i=0;i<dimensions.length;i++){
            l=dimensions[i][0];
            w=dimensions[i][1];
            if(l*l+w*w>res_d){
                res_d=l*l+w*w;
                res_a=l*w;
            }
            res_a=(l*l+w*w==res_d && l*w>res_a)?l*w:res_a;
        }
        return res_a;
    }
}

File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.py
class Solution(object):
    def areaOfMaxDiagonal(self, dimensions):
        res_d=2;
        res_a=1;
        for [l,w] in dimensions:
            if(l*l+w*w>res_d):
                res_d=l*l+w*w;
                res_a=l*w;
            elif(l*l+w*w==res_d):
                res_a=max(res_a,l*w);
        return res_a;
        """
        :type dimensions: List[List[int]]
        :rtype: int
        """